File: src/hello_java/d_while/Ex_1110.java
package hello_java.d_while;

import java.io.*;

/**
 * #1110 더하기 사이클
 *
 * 원래 수로 돌아올 때까지 연산을 반복하는 문제
 */

public class Ex_1110 {
    public static void main(String[] args) {
        try {
            BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
            BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

            int number = Integer.parseInt(br.readLine());
            int copy = number;
            int count = 0;

            do {
                number = ((number % 10) * 10) + (((number / 10) + (number % 10)) % 10);
                count++;
            } while (number != copy);

            bw.write(String.valueOf(count));
            bw.flush();
            bw.close();
            br.close();
        } catch (Exception e) {
            System.out.print(e.getMessage());
        }
    }
}

File: src/hello_kotlin/d_while/Ex_1110.kt
package hello_kotlin.d_while

import java.io.*

/**
 * #1110 더하기 사이클
 *
 * 원래 수로 돌아올 때까지 연산을 반복하는 문제
 */

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    val writer = BufferedWriter(OutputStreamWriter(System.out))

    var number = readLine().toInt()
    var copy = number
    var count = 0

    do {
        number = number % 10 * 10 + (number / 10 + number % 10) % 10
        count++
    } while (copy != number)

    writer.write(count.toString())
    writer.flush()
    writer.close()
    close()
}